src/Controllers/AnnouncementController.php

```diff
@@ -122,22 +127,6 @@ public function delete($id)
                 ]);
             }
```

Comment:
if you see the Announcement model, there has been a function `course` which should be able to be called by `$announcement->course` and will return the corresponding course for the announcement. I think instead of putting the join logic here (which is also quite expensive), we can utilize that one.

```php
            $announcement = Announcement::find($id);
            $course = $announcement->course;
            
            if (! ($announcement->created_by == $user->id || $course->created_by == $user->id) {
                throw ValidationException::withMessages([
                    'id' => 'Announcement not found',
                ]);
            }
```

Feel free to not implement it if you have another concern